Add cors capability to http-controller

fantasy0v0 opened the pull request:
#29

fantasy0v0 committed:
[controller] add cors capability to http-controller

fantasy0v0 commented:
Unit tests are still missing, and I will add them when I have time.

wkgcass commented:
建议把cors做成通用功能，放到http框架里
目前http框架支持request方向的chain，可以做Origin的校验
但是不支持response方向的chain。一个比较简单的方法是，在ctx中增加一个onResponse回调chain，在request方向注册回调，然后在headers发送前调用这组回调，这样即可添加response headers

这个基础功能实现后，可以跟jsonHandler类似，加一个corsHandler，专门检查Origin、添加Access-Control-Allow-Origin、处理Option请求

然后再在命令行添加cors相关配置

那个secret也应该做成一个单独的basicAuthHandler

fantasy0v0 committed:
[controller] rewrite cors handler

fantasy0v0 committed:
[controller] add CorsHandler

# Conflicts:
#	lib/src/main/java/io/vproxy/lib/http1/CoroutineHttp1ServerConnection.kt

fantasy0v0 committed:
[controller] add BasicAuthHandler

fantasy0v0 committed:
[controller] update doc

fantasy0v0 committed:
[controller] add cors and basicAuth test.

fantasy0v0 commented:
https://github.com/fantasy0v0/vproxy/actions/runs/11573596230/job/32216044558
